### PR TITLE
Backport the fixes related to replacing ApacheDS with UnboundID from pull #1783 to the master #1846 And Upgrade our UnboundID dependencies from 4.0.9 to 5.1.1 #1858

### PR DESCRIPTION
**Pull request by mknutsen79**
# Pull Request Description

This pull request closes strongbox/strongbox#1846

# Acceptance Test

* [ ] Building the code of the [`strongbox-parent`](https://github.com/strongbox/strongbox-parent) with `mvn clean install` still works.
* [ ] Building the code of the [`strongbox`](https://github.com/strongbox/strongbox) project with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [ ] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [ ] No

# Code Review And Pre-Merge Checklist

* [ ] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code in hard-to-understand areas.
* [ ] My changes generate no new warnings.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -604,6 +603,12 @@
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-compress</artifactId>
                 <version>${version.apache.commons.compress}</version>
+            </dependency>
+			<dependency>
```
Comment (carlspring) [inline pom.xml:607]: Incorrectly aligned.

Comment (mknutsen79) [reply pom.xml:607]: Hopefully I got that fixed. The alignment looked correct in my environment though so I fiddled with it to get it to be correct (hopefully).

Comment (carlspring) [reply pom.xml:607]: Well, (unless you haven't committed + pushed), then -- no, it's not fixed. Have a look at this diff:
```suggestion
		<dependency>
```

Could you please keep a closer look at the diffs when you push? (They are always displayed here and you can easily spot such cosmetic things. Thanks! :)

Comment (mknutsen79) [reply pom.xml:607]: I have no idea why its coming up looking like this i've worked on realigning a few times now. The pom is set how you want it to look on my PC so I have no current idea why its being pushed to look like this.

File: pom.xml [hunk 2]
```diff
@@ -604,6 +603,12 @@
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-compress</artifactId>
                 <version>${version.apache.commons.compress}</version>
+            </dependency>			
+			<dependency>
```
Comment (carlspring) [inline pom.xml:607]:
```suggestion
                <dependency>
```

File: pom.xml [hunk 3]
```diff
@@ -604,6 +603,12 @@
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-compress</artifactId>
                 <version>${version.apache.commons.compress}</version>
+            </dependency>			
+                <dependency>
```
Comment (carlspring) [inline pom.xml:607]:
```suggestion
            <dependency>
```

**Comment by carlspring** [comment]
Could you please rebase this as well? Thanks! :)

**Comment by carlspring** [comment]
Thank you for your efforts!

I will be closing this as a duplicate of #104 .

**Pull request closed without merging**